core/trino-main/src/main/java/io/trino/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -357,6 +367,12 @@ else if (pendingSplits.isEmpty()) {
                     throw new IllegalStateException("At least 1 split should have been scheduled for this plan node");
                 case SPLITS_ADDED:
                     state = State.NO_MORE_SPLITS;
+
+                    // todo: would be great to call getTableExecuteSplitsInfo if we are executing plan which requires that.
+                    TableExecuteContext tableExecuteContext = tableExecuteContextManager.getTableExecuteContextForQuery(stage.getStageId().getQueryId());
+                    Optional<List<Object>> tableExecuteSplitsInfo = splitSource.getTableExecuteSplitsInfo();
+                    tableExecuteSplitsInfo.ifPresent(tableExecuteContext::setSplitsInfo);
```

Comment:
doing it here seems wrong as it looks like we can have multiple SourcePartitionedSchedulers per single query. E.g. those can be created via `FixedSourcePartitionedScheduler`. Need a rework.
cc: @findepi